src/interactive-auth.js

```diff
@@ -143,19 +221,30 @@ InteractiveAuth.prototype = {
             prom = q.reject(e);
         }
 
-        prom.then(
+        prom = prom.then(
             function(result) {
                 console.log("result from request: ", result);
                 self._completionDeferred.resolve(result);
             }, function(error) {
-                if (error.httpStatus !== 401 || !error.data || !error.data.flows) {
+                // sometimes UI auth errors don't come with flows
+                const errorFlows = error.data ? error.data.flows : null;
+                const haveFlows = Boolean(self._data.flows) || Boolean(errorFlows);
+                if (error.httpStatus !== 401 || !error.data || !haveFlows) {
                     // doesn't look like an interactive-auth failure. fail the whole lot.
                     throw error;
                 }
-                self._data = error.data;
+                if (errorFlows) self._data = error.data;
                 self._startNextAuthStage();
             },
-        ).catch(this._completionDeferred.reject).done();
+        );
+        if (!ignoreFailure) {
```

Comment:
why is this necessary? surely we should only be ignoring interactive-auth failures, and they are caught further up?